### PR TITLE
Current progress

### PR DESCRIPTION
**Pull request by john0312**
Please let me know if you want to review them in smaller chunks, thank you.

### CODE REVIEW

File: kofserver/vm_manager.py [hunk 1]
```diff
@@ -81,7 +144,13 @@ def GetState(self):
 
     @staticmethod
     def LoadVM(vmPath):
-        pass
+        try:
+            with open(vmPath) as f:
+                result = yaml.load(f, Loader=yaml.FullLoader)
+        except Exception:
+            logging.exception("Failed to open vm config yml file %s"%vmpath)
```
Comment (gcobcqwe) [inline kofserver/vm_manager.py:151]: vmPath?

File: kofserver/scorer.py [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+        if self.game.state != GameState.GAME_RUNNING:
+            logging.warn("Trying to score players when game is not running")
+            return False
+        
+        assert self.pidLastCheckTime is not None
+        assert self.portLastCheckTime is not None
+        
+        pidUptime = {}
+        portUptime = {}
+        playerDict = {} # Just a dictionary to store the players
```
Comment (gcobcqwe) [inline kofserver/scorer.py:61]: Note: Just Relace Dict with Set structure if the users are unique?

Comment (john0312) [reply kofserver/scorer.py]: Done

File: kofserver/vm_manager.py (hunk 1 above)
Comment (john0312) [reply kofserver/vm_manager.py]: Done

**Comment by john0312** [comment]
Done, thanks for the reviews.

**Pull request merged**